customer_outstanding_statement/report/customer_outstanding_statement.py

```diff
@@ -0,0 +1,315 @@
+            END as b_1_30,
+            CASE
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is null THEN open_due
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is not null
+                                THEN open_due_currency
+                ELSE 0.0
+            END as b_30_60,
+            CASE
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is null THEN open_due
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is not null
+                                THEN open_due_currency
+                ELSE 0.0
+            END as b_60_90,
+            CASE
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is null THEN open_due
+                WHEN '%s' < date_maturity AND date_maturity <= '%s'
+                                AND currency_id is not null
+                                THEN open_due_currency
+                ELSE 0.0
+            END as b_90_120,
+            CASE
+                WHEN date_maturity <= '%s' AND currency_id is null
+                                THEN open_due
+                WHEN date_maturity <= '%s' AND currency_id is not null
+                                THEN open_due_currency
+                ELSE 0.0
+            END as b_over_120
+            FROM Q1
+            GROUP BY partner_id, currency_id, date_maturity, open_due,
+                                open_due_currency, move_id, company_id
+        """ % (today, today, minus_30, today, minus_30, today, minus_60,
+               minus_30, minus_60, minus_30, minus_90, minus_60, minus_90,
+               minus_60, minus_120, minus_90, minus_120, minus_90, minus_120,
+               minus_120)
+
+    def _show_buckets_sql_q3(self, company_id):
+        return """
+            SELECT Q2.partner_id, current, b_1_30, b_30_60, b_60_90, b_90_120,
+                                b_over_120,
+            COALESCE(Q2.currency_id, c.currency_id) AS currency_id
+            FROM Q2
+            JOIN res_company c ON (c.id = Q2.company_id)
+            WHERE c.id = %s
+        """ % company_id
+
+    def _show_buckets_sql_q4(self):
+        return """
+            SELECT partner_id, currency_id, sum(current) as current,
+                                sum(b_1_30) as b_1_30,
+                                sum(b_30_60) as b_30_60,
+                                sum(b_60_90) as b_60_90,
+                                sum(b_90_120) as b_90_120,
+                                sum(b_over_120) as b_over_120
+            FROM Q3
+            GROUP BY partner_id, currency_id
+        """
+
+    _bucket_dates = {
+        'today': fields.date.today(),
+        'minus_30': fields.date.today() - timedelta(days=30),
```

Comment:
A really killing nice to have (and not so difficult IMO) would be to be able to set up the time buckets